### PR TITLE
Remove castclass it should be unnecessary

### PR DESCRIPTION
**Pull request by Havunen**
ldelem_ref documentation says: 
Loads the element with an object reference at index onto the top of the stack as type O.

### CODE REVIEW

File: src/FastExpressionCompiler/FastExpressionCompiler.cs [hunk 1]
```diff
@@ -1579,11 +1579,7 @@ private static bool TryEmitIndex(IndexExpression expr, ILGenerator il)
 
                 if (expr.Arguments.Count == 1) // one dimensional array
                 {
-                    if (elemType.IsValueType())
```
Comment (Havunen) [inline src/FastExpressionCompiler/FastExpressionCompiler.cs]: I replaced this block of code with TryEmitArrayIndex method call to reduce code duplication

Comment (dadhi) [reply src/FastExpressionCompiler/FastExpressionCompiler.cs]: Ok.

**Comment by dadhi** [comment]
I merged another one, so closing this?

**Comment by Havunen** [comment]
yep lets close

**Pull request closed without merging**